src/handlers/monthly-post.js

```diff
@@ -4,13 +4,69 @@
  * A simple example includes a HTTP get method to get one item by id from a DynamoDB table.
  */
 
+function getLocalizedMonth(date, language) {
+  return date.toLocaleString(language, { month: 'long' });
+}
+
+function getLocalizedDate(date, language) {
+  return `${date.getDate()} (${date.toLocaleString(language, { weekday: 'short' })})`;
+}
+
+function getLocalizedMonthNames(dates, language) {
+  return localizedMonth = getLocalizedMonth(dates[0], language);
+}
+
+function getLocalizedListOfDatesInMonth(dates, language) {
+  // TODO: Check if all dates are same month
+  const localizedMonth = getLocalizedMonth(dates[0], language);
+  const localizedListOfDates = dates
+    .map(date => getLocalizedDate(date, language))
+    .join(', ');
+  return `${localizedMonth}: ${localizedListOfDates}`
+}
+
+function generateMonthlyPostVariables(pickupDates) {
+  // pickupDates: Sorted array of Date objects
+  // months: array of arrays, each containing the list of Date objects
+  // for one month's pickup dates. All elements are sorted
+  const firstMonth = pickupDates[0].getMonth();
+  const months = [
+    pickupDates.filter(date => date.getMonth() == firstMonth),
+    pickupDates.filter(date => date.getMonth() == (firstMonth + 1) % 12) // % in case of Dec-Jan
+  ];
+
+  return {
+    monthNames: {
+      en: months.map((date) => getLocalizedMonthNames(date, 'en')),
+      ja: months.map((date) => getLocalizedMonthNames(date, 'ja'))
+    },
+    datesListByMonth: {
+      en: months.map((date) => getLocalizedListOfDatesInMonth(date, 'en')),
+      ja: months.map((date) => getLocalizedListOfDatesInMonth(date, 'ja'))
+    }
+  };
+}
+
+function getPickupDatesFromQueryString(queryStringParams) {
+  // return a sorted array of Date objects
+  const pickupDatesParam = queryStringParams.pickup_dates;
+  const pickupDates = pickupDatesParam.split(',')
+    .map(dateString => new Date(dateString))
+    .sort((date1, date2) => date1.getTime() - date2.getTime());
+  // Note (Kai): I do not know why Array.sort() here needs a comparison function.
+  // From all my local tests, it sorted Date objects correctly, in ascending order.
+  // However, it went haywire when deployed, so I added an explicit comparison.
```

Comment:
Some difference between local and deployment environments...? Not sure what it would be though. 🤔

Reply:
Maybe. Even tried making the Node engine version the same (v14), didn't work.